cms/static/js/models/xblock_info.js

```diff
@@ -172,13 +182,33 @@ function(Backbone, _, str, ModuleUtils) {
             return !this.get('published') || this.get('has_changes');
         },
 
-        canBeDeleted: function(){
-            //get the type of xblock
-            if(this.get('override_type') != null) {
-                var type = this.get('override_type');
+        isDeletable: function() {
+            return this.isActionRequired('deletable');
+        },
+
+        isDraggable: function() {
+            return this.isActionRequired('draggable');
+        },
 
-                //hide/remove the delete trash icon if type is entrance exam.
-                if (_.has(type, 'is_entrance_exam') && type['is_entrance_exam']) {
+        isChildAddable: function(){
+            return this.isActionRequired('childAddable');
+        },
+
+        isHeaderVisible: function(){
+            if(this.get('is_header_visible')!= null) {
```

Comment:
I think I've updated the offending spots here -- there were only a couple from what I could see -- @asadiqbal08 FYI